pagopa/io-app
# fix(Cross): [IOAPPX-253] Wrong Danger messages in case of error

## Short description
This PR fixes a regression introduced by [this PR](https://github.com/pagopa/io-app/issues/5422) which prevents danger from displaying the proper message for PR without ticket numbers ([example](https://github.com/pagopa/io-app/pull/5296#issuecomment-1839032401)) or wrong ticket numbers ([example](https://github.com/pagopa/io-app/pull/4384#issuecomment-1431536463)).
Please notice that the whole script might be simplified drastically, thus we could refactor it in the future but the scope of these PRs is to remove the support for `pivotal`.

## List of changes proposed in this pull request
- Implements a check to verify if the result of `getJiraIdFromPrTitle` returns an empty array. If so, it is now explicitly returned, and subsequently utilized in `commentPrWithTicketsInfo` to send the appropriate message;
- If a ticket number is detected in the PR title, `getTicketsFromTitle` now directly returns the result of `getJiraTickets`. This adjustment aligns with the approach previously utilized [here](https://github.com/pagopa/io-app/pull/2935/files#diff-8ec5aa024b8a7cf5ead4ccf7f60297c3cdca64e13a74216867bc74914829a435R44), beside the mapping to a generic ticket type which is not needed anymore.

## How to test
Static checks but you can also check the history of the `pagopa-github-bot` message. As you can see I tested this without a ticket number, with a wrong ticket number, with multiple ticket numbers and also with the actual ticket number.
